### ISSUE TITLE
Install fails with a missing class

### ISSUE THREAD

**Issue by omnicolor** [captainhookphp/captainhook#252]
With a brand new repository, trying to run `./vendor/bin/captainhook install` fails with a missing class if you haven't done the configure step:

```bash
mordor captainhook-test(000 main) /$ composer require --dev captainhook/captainhook-phar
./composer.json has been updated
Running composer update captainhook/captainhook-phar
Loading composer repositories with package information
Updating dependencies
Lock file operations: 5 installs, 0 updates, 0 removals
  - Locking captainhook/captainhook-phar (5.23.3)
  - Locking phar-io/composer-distributor (1.0.2)
  - Locking phar-io/executor (1.0.1)
  - Locking phar-io/filesystem (2.0.1)
  - Locking phar-io/gnupg (1.0.2)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 5 installs, 0 updates, 0 removals
  - Installing phar-io/filesystem (2.0.1): Extracting archive
  - Installing phar-io/executor (1.0.1): Extracting archive
  - Installing phar-io/gnupg (1.0.2): Extracting archive
  - Installing phar-io/composer-distributor (1.0.2): Extracting archive
captainhook/captainhook-phar contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "captainhook/captainhook-phar" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] y
  - Installing captainhook/captainhook-phar (5.23.3): Extracting archive
  - Downloading artifact from https://github.com/captainhookphp/captainhook/releases/download/5.23.3/captainhook.phar
  - PHAR signature successfully verified
Generating autoload files
2 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
No security vulnerability advisories found.
Using version ^5.23 for captainhook/captainhook-phar
mordor captainhook-test(200 main) /$ ./vendor/bin/captainhook install

PHP Fatal error:  Uncaught Error: Class "Normalizer" not found in phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/vendor/symfony/string/AbstractUnicodeString.php:29
Stack trace:
#0 phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/vendor/symfony/console/Helper/Helper.php(45): [constant expression]()
#1 phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/vendor/symfony/console/Application.php(706): CaptainHook\Phar\Symfony\Component\Console\Helper\Helper::width()
#2 phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/vendor/symfony/console/Application.php(679): CaptainHook\Phar\Symfony\Component\Console\Application->doRenderThrowable()
#3 phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/vendor/symfony/console/Application.php(142): CaptainHook\Phar\Symfony\Component\Console\Application->renderThrowable()
#4 phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/vendor/symfony/console/Application.php(162): CaptainHook\Phar\Symfony\Component\Console\Application->CaptainHook\Phar\Symfony\Component\Console\{closure}()
#5 phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/bin/captainhook(63): CaptainHook\Phar\Symfony\Component\Console\Application->run()
#6 phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/bin/captainhook(64): CaptainHook\Phar\{closure}()
#7 /home/omni/code/captainhook-test/vendor/bin/captainhook(17): require('...')
#8 {main}
  thrown in phar:///home/omni/code/captainhook-test/vendor/bin/captainhook/vendor/symfony/string/AbstractUnicodeString.php on line 29
```

Using the non-.phar version tells the user to create the configuration file instead of dying.

**Comment by omnicolor** [comment]
Using `composer require --dev captainhook/captainhook` instead of the .phar require seems to work.

**Comment by sebastianfeldmann** [comment]
What PHP version are you running?

**Comment by omnicolor** [comment]
```
$ php -v
PHP 8.3.9 (cli) (built: Jul  5 2024 12:03:46) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.3.9, Copyright (c) Zend Technologies
    with Zend OPcache v8.3.9, Copyright (c), by Zend Technologies
    with Xdebug v3.3.2, Copyright (c) 2002-2024, by Derick Rethans
```

**Comment by sebastianfeldmann** [comment]
Could you run `php --info | grep intl` and check if your php is compiled with `'--enable-intl'` 

or you see lines like 

```
intl
intl.default_locale => no value => no value
intl.error_level => 0 => 0
intl.use_exceptions => Off => Off
```

**Comment by sebastianfeldmann** [comment]
My guess is, that your PHP comes without INTL extension and then the Symphony intl-polyfill should kick in. But since all Symfony classes get scoped by `Captainhook\Phar\Normalizer` the class `Normalizer` can not be found.

If you can confirm you do not have the `intl` extension I will try to figure out how I can solve the problem.

**Comment by omnicolor** [comment]
Yes, you're correct, my PHP doesn't have the `intl` extension.

**Comment by sebastianfeldmann** [comment]
Can you give the PHAR file from this beta release page a try?
https://github.com/captainhookphp/captainhook/releases/tag/5.23.4-beta

**Comment by omnicolor** [comment]
I'm not sure what's going on now, but the link on that release page seems a bit weird, showing the same version I got with `composer require --dev captainhook/captainhook-phar`:

```bash
$ wget --quiet https://github.com/captainhookphp/captainhook/releases/download/5.23.4-beta/captainhook.phar
$ php ./captainhook.phar --version
CaptainHook version 5.23.3 2024-07-07 #StandWithUkraine
$ ./vendor/bin/captainhook --version
CaptainHook version 5.23.3 2024-07-07 #StandWithUkraine
$ php ./captainhook.phar install

PHP Fatal error:  Uncaught Error: Class "Normalizer" not found in phar:///home/omni/code/slack/captainhook.phar/vendor/symfony/string/AbstractUnicodeString.php:29
Stack trace:
#0 phar:///home/omni/code/slack/captainhook.phar/vendor/symfony/console/Helper/Helper.php(57): [constant expression]()
#1 phar:///home/omni/code/slack/captainhook.phar/vendor/symfony/console/Application.php(731): CaptainHook\Phar\Symfony\Component\Console\Helper\Helper::width()
#2 phar:///home/omni/code/slack/captainhook.phar/vendor/symfony/console/Application.php(704): CaptainHook\Phar\Symfony\Component\Console\Application->doRenderThrowable()
#3 phar:///home/omni/code/slack/captainhook.phar/vendor/symfony/console/Application.php(142): CaptainHook\Phar\Symfony\Component\Console\Application->renderThrowable()
#4 phar:///home/omni/code/slack/captainhook.phar/vendor/symfony/console/Application.php(162): CaptainHook\Phar\Symfony\Component\Console\Application->CaptainHook\Phar\Symfony\Component\Console\{closure}()
#5 phar:///home/omni/code/slack/captainhook.phar/bin/captainhook(63): CaptainHook\Phar\Symfony\Component\Console\Application->run()
#6 phar:///home/omni/code/slack/captainhook.phar/bin/captainhook(64): CaptainHook\Phar\{closure}()
#7 /home/omni/code/slack/captainhook.phar(17): require('...')
#8 {main}
  thrown in phar:///home/omni/code/slack/captainhook.phar/vendor/symfony/string/AbstractUnicodeString.php on line 29
```

**Comment by sebastianfeldmann** [comment]
This seems to be a problem with the autoloading and the scoping for the PHAR file.
This is a bit try and error because for now I haven't setup an ENV without intl.
But you can try the lates alpha version https://github.com/captainhookphp/captainhook/releases/tag/5.23.4-alpha-1

**Comment by omnicolor** [comment]
Nope, that still doesn't seem to be working:

```bash
mordor /home/omni/code/captainhook-test$ wget https://github.com/captainhookphp/captainhook/releases/download/5.23.4-alpha-1/captainhook.phar
--2024-08-20 15:49:18--  https://github.com/captainhookphp/captainhook/releases/download/5.23.4-alpha-1/captainhook.phar
Resolving github.com (github.com)... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/64998184/3f3f9aa3-8900-434f-8e83-0507ddda481a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240820%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240820T154918Z&X-Amz-Expires=300&X-Amz-Signature=a0213a87d89464961133aff679f2ebb1e5e4307698cfcf29826323d113b13cba&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=64998184&response-content-disposition=attachment%3B%20filename%3Dcaptainhook.phar&response-content-type=application%2Foctet-stream [following]
--2024-08-20 15:49:18--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/64998184/3f3f9aa3-8900-434f-8e83-0507ddda481a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240820%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240820T154918Z&X-Amz-Expires=300&X-Amz-Signature=a0213a87d89464961133aff679f2ebb1e5e4307698cfcf29826323d113b13cba&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=64998184&response-content-disposition=attachment%3B%20filename%3Dcaptainhook.phar&response-content-type=application%2Foctet-stream
Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.111.133, 185.199.108.133, 185.199.110.133, ...
Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4597337 (4.4M) [application/octet-stream]
Saving to: ‘captainhook.phar’

captainhook.phar                  100%[===========================================================>]   4.38M  8.80MB/s    in 0.5s

2024-08-20 15:49:19 (8.80 MB/s) - ‘captainhook.phar’ saved [4597337/4597337]

mordor /home/omni/code/captainhook-test$ chmod +x captainhook.phar
mordor /home/omni/code/captainhook-test$ ./captainhook.phar --version
CaptainHook version 5.23.4-alpha 2024-07-07 #StandWithUkraine
mordor /home/omni/code/captainhook-test$ ./captainhook.phar install

PHP Fatal error:  Uncaught Error: Class "Normalizer" not found in phar:///home/omni/code/captainhook-test/captainhook.phar/vendor/symfony/string/AbstractUnicodeString.php:29
Stack trace:
#0 phar:///home/omni/code/captainhook-test/captainhook.phar/vendor/symfony/console/Helper/Helper.php(57): [constant expression]()
#1 phar:///home/omni/code/captainhook-test/captainhook.phar/vendor/symfony/console/Application.php(731): CaptainHook\Phar\Symfony\Component\Console\Helper\Helper::width()
#2 phar:///home/omni/code/captainhook-test/captainhook.phar/vendor/symfony/console/Application.php(704): CaptainHook\Phar\Symfony\Component\Console\Application->doRenderThrowable()
#3 phar:///home/omni/code/captainhook-test/captainhook.phar/vendor/symfony/console/Application.php(142): CaptainHook\Phar\Symfony\Component\Console\Application->renderThrowable()
#4 phar:///home/omni/code/captainhook-test/captainhook.phar/vendor/symfony/console/Application.php(162): CaptainHook\Phar\Symfony\Component\Console\Application->CaptainHook\Phar\Symfony\Component\Console\{closure}()
#5 phar:///home/omni/code/captainhook-test/captainhook.phar/bin/captainhook(66): CaptainHook\Phar\Symfony\Component\Console\Application->run()
#6 phar:///home/omni/code/captainhook-test/captainhook.phar/bin/captainhook(67): CaptainHook\Phar\{closure}()
#7 /home/omni/code/captainhook-test/captainhook.phar(17): require('...')
#8 {main}
  thrown in phar:///home/omni/code/captainhook-test/captainhook.phar/vendor/symfony/string/AbstractUnicodeString.php on line 29
mordor /home/omni/code/captainhook-test$ php --info | grep intl
mordor /home/omni/code/captainhook-test$
```

**Comment by apss-pohl** [comment]
Hi, 
i just did a fresh install and run into the same issue. Also it seems there is not intl for php8.3 in the current repos but it is available for php8.2
Tried " php-symfony-intl" but didnt change anything.

**Comment by omnicolor** [comment]
That fixes it. I obviously still get an error, but it's because *I* can't read the directions, not that the extension is missing!

```bash
mordor /home/omni/code/captainhook-test$ wget https://github.com/captainhookphp/captainhook/releases/download/5.23.4-alpha-2/captainhook.phar
--2024-08-21 21:50:08--  https://github.com/captainhookphp/captainhook/releases/download/5.23.4-alpha-2/captainhook.phar
Resolving github.com (github.com)... 140.82.113.3
Connecting to github.com (github.com)|140.82.113.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/64998184/9eae2976-2709-4d32-8ab6-f5c86ae0e509?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240821%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240821T215008Z&X-Amz-Expires=300&X-Amz-Signature=9ad70687a61f74b432ffd44ff6a834822f0e60d0c2e073f8daccf5ad8e65274b&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=64998184&response-content-disposition=attachment%3B%20filename%3Dcaptainhook.phar&response-content-type=application%2Foctet-stream [following]
--2024-08-21 21:50:08--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/64998184/9eae2976-2709-4d32-8ab6-f5c86ae0e509?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240821%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240821T215008Z&X-Amz-Expires=300&X-Amz-Signature=9ad70687a61f74b432ffd44ff6a834822f0e60d0c2e073f8daccf5ad8e65274b&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=64998184&response-content-disposition=attachment%3B%20filename%3Dcaptainhook.phar&response-content-type=application%2Foctet-stream
Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.109.133, 185.199.111.133, 185.199.108.133, ...
Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.109.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2245312 (2.1M) [application/octet-stream]
Saving to: ‘captainhook.phar’

captainhook.phar                  100%[===========================================================>]   2.14M  --.-KB/s    in 0.1s

2024-08-21 21:50:09 (15.0 MB/s) - ‘captainhook.phar’ saved [2245312/2245312]

mordor /home/omni/code/captainhook-test$ chmod +x captainhook.phar
mordor /home/omni/code/captainhook-test$ ./captainhook.phar --version
CaptainHook version 5.23.4-alpha-1 2024-07-07 #StandWithUkraine
mordor /home/omni/code/captainhook-test$ ./captainhook.phar install

In ConfigAware.php line 53:

  Please create a captainhook configuration first
  Run 'captainhook configure'
  If you have a configuration located elsewhere use the --configuration option


install [-c|--configuration [CONFIGURATION]] [-g|--git-directory [GIT-DIRECTORY]] [--only-enabled] [-f|--force] [-s|--skip-existing] [--move-existing-to [MOVE-EXISTING-TO]] [-b|--bootstrap [BOOTSTRAP]] [-m|--run-mode [RUN-MODE]] [-e|--run-exec [RUN-EXEC]] [-p|--run-path [RUN-PATH]] [--] [<hook>]

mordor /home/omni/code/captainhook-test$ php --info | grep intl
mordor /home/omni/code/captainhook-test$
```

And obviously, it still works if I install the intl extension. Fixed in [5.23.4-alpha-2](https://github.com/captainhookphp/captainhook/releases/tag/5.23.4-alpha-2).

**Comment by sebastianfeldmann** [comment]
Great that it works now.
And thanks a lot for the donation, it's greatly appreciated ❤️
